src/components/WwwFrame/Header/KvAtbModalContainer.vue

```diff
@@ -144,7 +188,32 @@ const fetchPostCheckoutAchievements = async loanIds => {
 	}).then(({ data }) => {
 		const loanAchievements = data.postCheckoutAchievements?.overallProgress ?? [];
 		contributingAchievements.value = loanAchievements.filter(achievement => achievement.postCheckoutTier !== achievement.preCheckoutTier); // eslint-disable-line max-len
-		showModalContent.value = contributingAchievements.value.length;
+		const nonContributingAchievements = loanAchievements.filter(achievement => achievement.postCheckoutTier === achievement.preCheckoutTier); // eslint-disable-line max-len
+
+		const filteredAchievementsData = [];
+		nonContributingAchievements.forEach(achievement => {
+			const filteredAchievement = badgeAchievementData.value.find(badgeData => achievement.achievementId === badgeData.id); // eslint-disable-line max-len
+			const activeTier = filteredAchievement?.tiers?.find(tier => !tier.completedDate);
+			filteredAchievementsData.push({ ...filteredAchievement, ...activeTier });
+		});
+
+		filteredAchievementsData.sort((a, b) => defaultBadges.indexOf(a.id) - defaultBadges.indexOf(b.id));
+		const oneLoanAwayAchievement = filteredAchievementsData.find(achievement => {
+			return (achievement.target - 1) - achievement.totalProgressToAchievement === 1;
+		});
+		if (oneLoanAwayAchievement) {
+			oneLoanAwayFilteredUrl.value = getFilteredUrl(oneLoanAwayAchievement);
+			oneLoanAwayCategory.value = categoryNames[oneLoanAwayAchievement.id];
+			const { target } = oneLoanAwayAchievement;
+			oneAwayText.value = `${target - 1} of ${target}`;
+			showModalContent.value = true;
+			modalVisible.value = true;
+			return;
+		}
+		if (addedLoan.value?.basketSize < BASKET_LIMIT_SIZE_FOR_EXP) {
```

Comment:
Might be clearer to make this an `else if` instead of returning in the other if statement. Thoughts?

Reply:
Oh yeah, makes sense, thanks!